src/types.ts

```diff
@@ -1,15 +1,277 @@
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminGetOneParams extends GetOneParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminGetManyParams extends GetManyParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminGetManyReferenceParams
+  extends GetManyReferenceParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminUpdateParams extends UpdateParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminUpdateManyParams extends UpdateManyParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminCreateParams extends CreateParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminDeleteParams extends DeleteParams {
+  searchParams?: SearchParams;
+}
+
+export interface ApiPlatformAdminDeleteManyParams extends DeleteManyParams {
+  searchParams?: SearchParams;
+}
+
+export type ApiPlatformAdminDataProviderParams =
+  | ApiPlatformAdminGetListParams
+  | ApiPlatformAdminGetOneParams
+  | ApiPlatformAdminGetManyParams
+  | ApiPlatformAdminGetManyReferenceParams
+  | ApiPlatformAdminUpdateParams
+  | ApiPlatformAdminUpdateManyParams
+  | ApiPlatformAdminCreateParams
+  | ApiPlatformAdminDeleteParams
+  | ApiPlatformAdminDeleteManyParams;
+export type DataProviderType =
+  | typeof GET_LIST
+  | typeof GET_ONE
+  | typeof GET_MANY
+  | typeof GET_MANY_REFERENCE
+  | typeof UPDATE
+  | typeof UPDATE_MANY
+  | typeof CREATE
+  | typeof DELETE
+  | typeof DELETE_MANY;
+export type ApiPlatformAdminDataProviderTypeParams<T extends DataProviderType> =
+  T extends typeof GET_LIST
+    ? ApiPlatformAdminGetListParams
+    : T extends typeof GET_ONE
+    ? ApiPlatformAdminGetOneParams
+    : T extends typeof GET_MANY
+    ? ApiPlatformAdminGetManyParams
+    : T extends typeof GET_MANY_REFERENCE
+    ? ApiPlatformAdminGetManyReferenceParams
+    : T extends typeof UPDATE
+    ? ApiPlatformAdminUpdateParams
+    : T extends typeof UPDATE_MANY
+    ? ApiPlatformAdminUpdateManyParams
+    : T extends typeof CREATE
+    ? ApiPlatformAdminCreateParams
+    : T extends typeof DELETE
+    ? ApiPlatformAdminDeleteParams
+    : T extends typeof DELETE_MANY
+    ? ApiPlatformAdminDeleteManyParams
+    : never;
+
+export interface HydraDataProviderFactoryParams {
+  entrypoint: string;
+  httpClient?: (
+    url: URL,
+    options?: HydraHttpClientOptions,
+  ) => Promise<HydraHttpClientResponse>;
+  apiDocumentationParser?: (
+    entrypointUrl: string,
+    options?: ApiDocumentationParserOptions,
+  ) => Promise<ApiDocumentationParserResponse>;
+  mercure?: Partial<MercureOptions>;
+  useEmbedded?: boolean;
+  disableCache?: boolean;
+}
 
 export interface ApiPlatformAdminDataProvider extends DataProvider {
+  getList: <RecordType extends ApiPlatformAdminRecord>(
+    resource: string,
+    params: ApiPlatformAdminGetListParams,
+  ) => Promise<GetListResult<RecordType>>;
```

Comment:
you could create a type to avoid repeating all this, for example:

`type ApiPlatformDataProviderType<T> = RecordType extends ApiPlatformAdminRecord>(
    resource: string,
    params: ApiPlatformAdminGetListParams,
  ) => Promise<T<RecordType>>'

```
getList: ApiPlatformDataProviderType<GetListResult>
```

Reply:
It's trickier than that: the result needs to use a generic type so it can't be easily replaced.